integration/report_with_rule_updates_test.go

```diff
@@ -0,0 +1,235 @@
+// +build integration
+
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3cluster/kv/mem"
+	"github.com/m3db/m3metrics/metric/id"
+	"github.com/m3db/m3metrics/metric/unaggregated"
+	"github.com/m3db/m3metrics/policy"
+
+	"github.com/stretchr/testify/require"
+)
+
+func testReportWithRuleUpdates(
+	t *testing.T,
+	testDesc string,
+	intputIDGen idGenerator,
+	outputRes outputResults,
+	iterPool id.SortedTagIteratorPool,
+) {
+
+	namespaces := defaultNamespaces()
```

Comment:
Would it be worthwhile to move these into the function signature? When looking at the integration tests, such as `TestReportMatchMappingWithRuleUpdates `, it's unclear from the test why the id's match a given mapping or rollup rule.

Reply:
Sure thing